cloud/scope/machine.go

```diff
@@ -87,6 +88,25 @@ type MachineScope struct {
 	AzureMachine *infrav1.AzureMachine
 }
 
+// VMSpecs returns the VM specs.
+func (m *MachineScope) VMSpecs() []azure.VMSpec {
```

Comment:
why does this return an array?

Reply:
I actually went back and forth on this but in the end I decided to stick with an array because 1) It's consistent with the others 2) I could see a scenario where we support a VM jumpbox / "bastion" in the future for ssh and it would require creating 2 VMs per AzureMachine.

Reply:
2 VMs per AzureMachine? That feels aggressive...I would imagine something like 1 per cluster (or perhaps 1 per pool if same subnet jumpbox is required). 

I don't see consistency as a super strong argument personally, but given the bastion scenario I could see us reusing the same service so I guess it makes sense.